components/automate-cli/pkg/verifyserver/services/stopmockserverservice/stopmockserverservice_test.go

```diff
@@ -0,0 +1,189 @@
+			err:      "Error while shutting UDP server",
+		},
+		{
+			name:     "Error while stopping TCP server",
+			port:     1224,
+			protocol: "tcp",
+			cert:     "",
+			key:      "",
+			wantErr:  true,
+			err:      "Error while shutting TCP server",
+		},
+		{
+			name:     "Stop HTTPS server",
+			port:     80,
+			protocol: "https",
+			cert:     SERVER_CERT,
+			key:      SERVER_KEY,
+			wantErr:  true,
+			err:      "Error while shutting HTTPs server",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := stopmockserverservice.NewStopMockServerService(l)
+			if tt.wantErr {
+				server := ErrorForServer(tt.protocol, tt.port)
+				err = s.StopMockServer(server)
+				time.Sleep(100 * time.Millisecond)
+				assert.Error(t, err)
+				assert.Equal(t, tt.err, err.Error())
+				assert.Contains(t, cw.Output(), tt.err)
+			} else {
+				server, err := StartServer(tt.protocol, tt.port, tt.cert, tt.key, l)
+				assert.NoError(t, err)
+				err = s.StopMockServer(server)
+				time.Sleep(100 * time.Millisecond)
+				assert.NoError(t, err)
+				assert.Contains(t, cw.Output(), tt.out)
+			}
+		})
+	}
+}
+
+func StartServer(protocol string, port int, cert string, key string, l logger.Logger) (*models.Server, error) {
+	servers := startmockserverservice.New(l)
+
+	cfg := &models.StartMockServerRequestBody{
+		Protocol: protocol,
+		Port:     port,
+		Cert:     cert,
+		Key:      key,
+	}
+	err := servers.StartMockServer(cfg)
+	if err != nil {
+		return nil, err
+	}
+	server := servers.GetMockServers()[0]
+	return server, nil
+}
+
+func ErrorForServer(protocol string, port int) *models.Server {
+	signalChan := make(chan bool, 1)
+	return &models.Server{
+		Port:         port,
+		ListenerTCP:  &MockTCPListener{},
+		ListenerUDP:  &MockUDPListener{},
+		ListenerHTTP: &MockHTTPSServer{},
+		SignalChan:   signalChan,
+		Protocol:     protocol,
+	}
+}
+
+type MockTCPListener struct{}
+
+func (l *MockTCPListener) Accept() (net.Conn, error) {
+	return nil, nil
+}
+
+func (l *MockTCPListener) Close() error {
+	return errors.New("Error while shutting TCP server")
+}
+
+func (l *MockTCPListener) Addr() net.Addr {
+	return nil
+}
+
+type MockUDPListener struct{}
+
+func (l *MockUDPListener) Close() error {
+	return errors.New("Error while shutting UDP server")
+}
+
+func (l *MockUDPListener) ReadFrom(p []byte) (n int, addr net.Addr, err error) {
+	return 0, nil, nil
+}
+
+func (l *MockUDPListener) WriteTo(p []byte, addr net.Addr) (n int, err error) {
+	return 0, nil
+}
+
+func (l *MockUDPListener) LocalAddr() net.Addr {
+	return nil
+}
+
+func (l *MockUDPListener) SetDeadline(t time.Time) error {
+	return nil
+}
+
+func (l *MockUDPListener) SetReadDeadline(t time.Time) error {
+	return nil
+}
+
+func (l *MockUDPListener) SetWriteDeadline(t time.Time) error {
+	return nil
+}
+
+type MockHTTPSServer struct{}
+
+func (s *MockHTTPSServer) Shutdown(ctx context.Context) error {
+	return errors.New("Error while shutting HTTPs server")
+}
```

Comment:
Can you create these mocks similar to how we create mocks for service so that it can be reusable

Reply:
Also move the mocks to outside lib folder so that it could be used across the app

Reply:
done